aws/resource_aws_kms_grant.go

```diff
@@ -0,0 +1,502 @@
+package aws
+
+import (
+	"bytes"
+	"fmt"
+	"log"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/kms"
+	"github.com/hashicorp/terraform/helper/hashcode"
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/helper/schema"
+)
+
+func resourceAwsKmsGrant() *schema.Resource {
+	return &schema.Resource{
+		// There is no API for updating/modifying grants, hence no Update
+		// Instead changes to most fields will force a new resource
+		Create: resourceAwsKmsGrantCreate,
+		Read:   resourceAwsKmsGrantRead,
+		Delete: resourceAwsKmsGrantDelete,
+		Exists: resourceAwsKmsGrantExists,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ForceNew:     true,
+				ValidateFunc: validateAwsKmsGrantName,
+			},
+			"key_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"grantee_principal": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validateArn,
+			},
+			"operations": {
+				Type: schema.TypeSet,
+				Set:  schema.HashString,
+				Elem: &schema.Schema{
+					Type:         schema.TypeString,
+					ValidateFunc: validateAwsKmsGrantOperation,
```

Comment:
We can avoid a custom validation function and associated unit test by using Terraform core's `validation.StringInSlice()` 😄  (it should also use SDK provided constants)

```go
ValidateFunc: validation.StringInSlice([]string{
  kms.GrantOperationCreateGrant,
  kms.GrantOperationDecrypt,
  kms.GrantOperationDescribeKey,
  kms.GrantOperationEncrypt,
  kms.GrantOperationGenerateDataKey,
  kms.GrantOperationGenerateDataKeyWithoutPlaintext,
  kms.GrantOperationReEncryptFrom,
  kms.GrantOperationReEncryptTo,
  kms.GrantOperationRetireGrant,
}, false),
```

Reply:
Done.